apisix/balancer/ewma.lua

```diff
@@ -135,16 +153,23 @@ local function _ewma_find(ctx, up_nodes)
         return nil, 'up_nodes trans error'
     end
 
+    local endpoint, backendpoint = peers[1], nil
+
     if #peers > 1 then
-        endpoint = pick_and_score(peers)
-    else
-        endpoint = peers[1]
+        local a, b = math.random(1, #peers), math.random(1, #peers - 1)
+        if b >= a then
+            b = b + 1
+        end
+
+        endpoint, backendpoint = peers[a], peers[b]
+        if score(endpoint) > score(backendpoint) then
+            endpoint, backendpoint = backendpoint, endpoint
+        end
```

Comment:
We need to sync the `tried_endpoint` check from https://github.com/kubernetes/ingress-nginx/blob/a2e77185cc2e91278962f4f1267246c8fefc6e73/rootfs/etc/nginx/lua/balancer/ewma.lua#L180 to our new implementation.
You can take a look at:
https://github.com/apache/apisix/blob/bbbdf58d555ee89144b6923e751deabf7e0bbf15/apisix/balancer/roundrobin.lua#L54-L74

Reply:
Maybe this is another topic we could improve the stability, like passive/active healthcheck, tried record & tries count?

Reply:
It would be better to fix this known issue in this PR.

Reply:
Got it

Reply:
It seems this problem is still not addressed?

Reply:
Sorry, filter logic is added, I'm focusing something else in these days.